framework/src/functions/PiecewiseBase.C

```diff
@@ -16,65 +17,156 @@ PiecewiseBase::validParams()
 {
   InputParameters params = Function::validParams();
 
-  MooseEnum axis("x=0 y=1 z=2 0=3 1=4 2=5");
-  axis.deprecate("0", "x");
-  axis.deprecate("1", "y");
-  axis.deprecate("2", "z");
+  MooseEnum axis("x=0 y=1 z=2");
   params.addParam<MooseEnum>(
       "axis", axis, "The axis used (x, y, or z) if this is to be a function of position");
+  params.addParam<std::vector<Real>>("xy_data",
+                                     "All function data, supplied in abscissa, ordinate pairs");
+  params.addParam<std::vector<Real>>("x", "The abscissa values");
+  params.addParam<std::vector<Real>>("y", "The ordinate values");
+  params.addParam<FileName>("data_file", "File holding CSV data for use with Piecewise");
```

Comment:
Aie! How did these sneak in...I created `PiecewiseBase` so I could make a custom linear function with inputs not like `x` and `y`, but rather `ramp_rates`, and `hold_times`, while still allowing me to pass it to `IterationAdaptiveDT`, which requires a `PiecewiseBase`. I think I need to make a `PiecewiseBaseBase` now :(

It looks like `PiecewiseBase` is only used in `PiecewiseConstant` and `PiecewiseLinearBase`, at least in MOOSE. Maybe I can just move these input parameters into those two files...

Reply:
Yeah, you took away `Piecewise` which gave me the flexibility I needed to not rely on input parameters to build my interpolation @dschwen!

Reply:
@topher, is it maybe this PR? Was the problem toy raised ever solved?

Reply:
Yeah, I thought I fixed this...

Reply:
Also, that isn't me, @dschwen ...sorry @topher

Reply:
That's what he deserves for snatching that username away from you :-D

Reply:
Right! Doesn't seem to be active either. :)